mono/mini/genmdesc.c

```diff
@@ -220,7 +220,7 @@ init_table (void) {
 
 static void
 output_char (FILE *f, char c) {
-	if (isalnum (c))
+	if (isalnum ((unsigned char)c)) // Cast to unsigned char to work around a bug in MSVCRT in debug builds
```

Comment:
`int isalnum (int)`  - so there's a numeric conversion from `char` to `int`.  Since it's just plain `char` rather than `signed char` or `unsigned char` (and `char` defaults to `signed` on MSVC) you get -1.  Actually the default is (after some googling) signed on gcc and clang also (unless `-funsigned-char`) except on Android.  So why do we only notice this on Windows?